### PR TITLE
Exampledesigns

### PR DESCRIPTION
**Pull request by csoneson**

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -10,7 +10,9 @@ Authors@R: c(person("Charlotte", "Soneson", role = c("aut", "cre"),
     comment = c(ORCID = "0000-0003-3252-7758")),
     person("Michael", "Love", role = "aut", 
     email = "michaelisaiahlove@gmail.com",
-    comment = c(ORCID = "0000-0001-8401-0545")))
+    comment = c(ORCID = "0000-0001-8401-0545")),
+    person("Florian", "Geier", role = "aut",
```
Comment (federicomarini) [inline DESCRIPTION]: Does Florian also have a ORCID?

File: NAMESPACE [hunk 2]
```diff
@@ -69,8 +79,15 @@ importFrom(shinydashboard,menuItem)
 importFrom(shinydashboard,notificationItem)
 importFrom(shinydashboard,valueBox)
 importFrom(stats,as.formula)
+importFrom(stats,cor)
+importFrom(stats,cov2cor)
+importFrom(stats,lm)
 importFrom(stats,model.matrix)
 importFrom(stats,relevel)
+importFrom(stats,var)
+importFrom(tibble,rownames_to_column)
```
Comment (federicomarini) [inline NAMESPACE]: Can we "avoid" depending on `tibble`? Nothing against it per se - more for the "we just would need that function".

File: R/exploreModelMatrix.R [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
 #' Given a sample table and a design formula, explore the resulting design
-#' matrix graphically in a shiny app.
+#' matrix graphically in an interactive application.
 #'
 #' @param sampleData A \code{data.frame} with sample information.
 #' @param designFormula A \code{formula}. All components of the terms must be
 #'   present as columns in \code{sampleData}.
 #'
-#' @author Charlotte Soneson
+#' @author Charlotte Soneson, Federico Marini
```
Comment (federicomarini) [inline R/exploreModelMatrix.R]: Florian and Mike here as well?

File: NAMESPACE (hunk 2 above)
Comment (federicomarini) [reply NAMESPACE]: Ok, somewhat answering to myself: we already have dplyr as dep, so this one is quite harmless

File: R/exploreModelMatrix.R [hunk 4]
```diff
@@ -363,17 +502,189 @@ ExploreModelMatrix <- function(sampleData = NULL, designFormula = NULL) {
...
+    # Plot correlation among coefficients -------------------------------------
+    output$correlation_matrix_plot <- shiny::renderPlot({
+      shiny::validate(
+        shiny::need(
+          input$designformula != "" &&
+            .isValidFormula(as.formula(input$designformula), values$sampledata),
+          paste0("Please provide a formula where all terms appear in ",
+                 "the sample data")
+        )
+      )
```
Comment (federicomarini) [inline R/exploreModelMatrix.R]: General thought: do we want to provide validate-need messages "more to the point"? Or is it fine to leave the generic one like it is now?

File: vignettes/exploreModelMatrix.Rmd [hunk 5]
```diff
@@ -1,10 +1,8 @@
 ---
 title: "ExploreModelMatrix"
-author: "Charlotte Soneson"
+author: "Charlotte Soneson, Federico Marini and Michael I Love"
```
Comment (federicomarini) [inline vignettes/exploreModelMatrix.Rmd]: same as above: Florian to be included or is the vignette "a product apart"?

**Comment by federicomarini** [comment]
Wondering whether it is worth to get below 5Mb for the package size, this would give us a NOTE-free R CMD check. Maybe some figures with smaller size in the vignette?

File: DESCRIPTION (hunk 1 above)
Comment (csoneson) [reply DESCRIPTION]: Added

File: NAMESPACE (hunk 2 above)
Comment (csoneson) [reply NAMESPACE]: Either is fine with me, but I rewrote the two instances using tibble and removed it.

File: R/exploreModelMatrix.R (hunk 3 above)
Comment (csoneson) [reply R/exploreModelMatrix.R]: Added

File: vignettes/exploreModelMatrix.Rmd (hunk 5 above)
Comment (csoneson) [reply vignettes/exploreModelMatrix.Rmd]: Added

File: R/exploreModelMatrix.R (hunk 4 above)
Comment (csoneson) [reply R/exploreModelMatrix.R]: I think that might be useful. Maybe open an issue so that we remember

**Comment by federicomarini** [comment]
Quick extra thought: I thought it could be nice to have all figures centered in the vignette (especially since the text is nicely taking the central stripe as well).
Thoughts/anything against it?

**Pull request merged**